dotnet/src/Functions/Functions.OpenApi/Extensions/OpenApiKernelExtensions.cs

```diff
@@ -333,13 +346,24 @@ async Task<RestApiOperationResponse> ExecuteAsync(KernelArguments variables, Can
 
         var returnParameter = operation.GetDefaultReturnParameter();
 
+        // Add unstructured metadata, specific to Open API, to the metadata property bag.
+        var metadataPropertyBag = new Dictionary<string, object?>();
+        metadataPropertyBag.Add(OpenApiKernelExtensions.OperationExtensionsIdKey, operation.Id);
+        metadataPropertyBag.Add(OpenApiKernelExtensions.OperationExtensionsPathKey, operation.Path);
+        metadataPropertyBag.Add(OpenApiKernelExtensions.OperationExtensionsMethodKey, operation.Method.ToString());
+        if (operation.Extensions is not null && operation.Extensions.Any())
```

Comment:
```suggestion
        if (operation.Extensions is { Count: > 0 })
```
nit: the new pattern matching syntaxis that checks that the collection is initialized and contains elements.